packages/@jsii/check-node/src/constants.ts

```diff
@@ -1,69 +1,161 @@
+      (majorVersion) => new NodeRelease(majorVersion, { endOfLife: true }),
+    ),
+
+    // Past end-of-life releases
+    new NodeRelease(13, { endOfLife: new Date('2020-06-01') }),
+    new NodeRelease(15, { endOfLife: new Date('2021-06-01') }),
+
+    // Deprecated releases
+    new NodeRelease(12, {
+      endOfLife: new Date('2022-04-30'),
+      supportedRange: '^12.7.0',
+    }),
+
+    // Currently active releases
+    new NodeRelease(14, {
+      endOfLife: new Date('2023-04-30'),
+      supportedRange: '^14.5.0',
+    }),
+    new NodeRelease(16, {
+      endOfLife: new Date('2024-04-30'),
+      supportedRange: '^16.3.0',
+    }),
+    new NodeRelease(17, {
+      endOfLife: new Date('2022-06-01'),
+      supportedRange: '^17.3.0',
+    }),
+    new NodeRelease(18, { endOfLife: new Date('2025-04-30') }),
+
+    // Future (planned releases)
+    new NodeRelease(19, { endOfLife: new Date('2023-06-01'), pending: true }),
+    new NodeRelease(20, { endOfLife: new Date('2026-04-30'), pending: true }),
+  ];
 
   /**
-   * EOL releases are unsupported, so this will emit a SCARY RED WARNING.
+   * @returns the `NodeRelease` corresponding to the version of the node runtime
+   *          executing this code (as provided by `process.version`), and a
+   *          boolean indicating whether this version is known to be broken. If
+   *          the runtime does not correspond to a known node major, this
+   *          returns and `undefined` release object.
    */
-  END_OF_LIFE = 'end-of-life',
+  public static forThisRuntime(): {
+    nodeRelease: NodeRelease | undefined;
+    knownBroken: boolean;
+  } {
+    const semver = new SemVer(version);
+    const majorVersion = semver.major;
+
+    for (const nodeRelease of this.ALL_RELEASES) {
+      if (nodeRelease.majorVersion === majorVersion) {
+        return {
+          nodeRelease,
+          knownBroken: !nodeRelease.supportedRange.test(semver),
+        };
+      }
+    }
+
+    return { nodeRelease: undefined, knownBroken: false };
+  }
 
   /**
-   * Not end-of-life, but also not supported. This is usually the case for early
-   * releases on a new major train, where some of the features are not available
-   * yet. These emit a SCARY RED WARNING, but the messaging is slightly
-   * different than that of END_OF_LIFE.
+   * The major version of this node release.
    */
-  UNSUPPORTED = 'unsupported',
+  public readonly majorVersion: number;
 
   /**
-   * Those releases are newer than the current software package, so we have not
-   * tested with it. People may encounter all kinds of weird bugs, especially
-   * early in the release cycle... So we'll issue a yellow warning here.
+   * The date on which this release range starts to be considered end-of-life.
+   * Can also be `true` to denote that this is already end-of-life, without
+   * specifying a particular date.
    */
-  UNTESTED = 'untested',
-}
+  public readonly endOfLifeDate: Date | undefined;
```

Comment:
The docstring says this can be "true" but it's typed as `Date | undefined`. Bug?

Reply:
No it's just I changed my mind twice here but I forgot to update the text...